tests/test_microgrid/test_timeout.py

```diff
@@ -0,0 +1,185 @@
+        time.sleep(GRPC_SERVER_DELAY)
+        return ComponentList(components=[])
+
+    mocker.patch.object(servicer, "ListComponents", mock_list_components)
+    server = MockGrpcServer(servicer, port=57809)
+    await server.start()
+
+    target = "[::]:57809"
+    grpc_channel = grpc.aio.insecure_channel(target)
+    client = MicrogridGrpcClient(grpc_channel=grpc_channel, target=target)
+
+    with pytest.raises(grpc.aio.AioRpcError) as err_ctx:
+        _ = await client.components(timeout=GRPC_CALL_TIMEOUT)
+    assert err_ctx.value.code() == grpc.StatusCode.DEADLINE_EXCEEDED
+    await server.wait_for_termination(GRPC_SERVER_SHUTDOWN_DELAY)
+
+
+async def test_connections_timeout(mocker: MockerFixture) -> None:
+    """Test if the connections() method properly raises AioRpcError"""
+    servicer = MockMicrogridServicer()
+
+    # pylint: disable=unused-argument
+    def mock_list_connections(
+        request: ConnectionFilter, context: Any
+    ) -> ConnectionList:
+        time.sleep(GRPC_SERVER_DELAY)
+        return ConnectionList(connections=[])
+
+    mocker.patch.object(servicer, "ListConnections", mock_list_connections)
+    server = MockGrpcServer(servicer, port=57809)
+    await server.start()
+
+    target = "[::]:57809"
+    grpc_channel = grpc.aio.insecure_channel(target)
+    client = MicrogridGrpcClient(grpc_channel=grpc_channel, target=target)
+
+    with pytest.raises(grpc.aio.AioRpcError) as err_ctx:
+        _ = await client.connections(timeout=GRPC_CALL_TIMEOUT)
+    assert err_ctx.value.code() == grpc.StatusCode.DEADLINE_EXCEEDED
+    await server.wait_for_termination(GRPC_SERVER_SHUTDOWN_DELAY)
+
+
+async def test_set_bound_timeout(mocker: MockerFixture) -> None:
+    """Test if the set_bounds() method properly raises AioRpcError"""
+    servicer = MockMicrogridServicer()
+
+    # pylint: disable=unused-argument
+    def mock_set_bounds(
+        request_iterator: Iterator[SetBoundsParam], context: Any
+    ) -> Empty:
+        time.sleep(GRPC_SERVER_DELAY)
+        return Empty()
+
+    mocker.patch.object(servicer, "SetBounds", mock_set_bounds)
+    server = MockGrpcServer(servicer, port=57809)
+    await server.start()
+
+    target = "[::]:57809"
+    grpc_channel = grpc.aio.insecure_channel(target)
+    client = MicrogridGrpcClient(grpc_channel=grpc_channel, target=target)
+
+    with pytest.raises(grpc.aio.AioRpcError) as err_ctx:
+        # To be investigated: passing a timeout a little bit smaller than the sleep
+        # time in `mock_set_bounds` should be sufficient, but it doesn't work. Making
+        # the timeout much smaller than the sleep time does work. E.g.
+        # sleep => 0.3, timeout = 0.1 or even 0.001 doesn't work, but 0.00001 works
+        await client.set_bounds(component_id=1, lower=0.0, upper=10.0, timeout=0.00001)
+    assert err_ctx.value.code() == grpc.StatusCode.DEADLINE_EXCEEDED
+    await server.stop(GRPC_SERVER_SHUTDOWN_DELAY)
```

Comment:
So what's the error here? It's not finishing after `timeout` with `DEADLINE_EXCEEDED`? Or is there any other error. In any case, I'm not familiar with gRPC, so I would have to start debugging from scratch to help here. Not sure if @ela-kotulska-frequenz or @sahas-subramanian-frequenz are more familiar and can help.

Tests are still failing though, not sure if because of this test or something else (haven't checked the logs).

Reply:
> await client.set_bounds(component_id=1, lower=0.0, upper=10.0, timeout=0.00001

What if the call finish in this short timeout? I think we should avoid this kind of tests...

Reply:
Do you mean `set_bounds` on the `client` instance? If I mock `set_bounds` then there will be no gRPC call.

I am keeping the client as it is, but only mocking the server response to take long enough to trigger a timeout error in the client.

Reply:
@leandro-lucarella-frequenz 
> So what's the error here? It's not finishing after `timeout` with `DEADLINE_EXCEEDED`?

Yes, exactly. It says `did not raise`. I cannot find out why, because I am mocking the `SetBounds()` method on the server instance to sleep for `0.3`:
```
    def mock_set_bounds(
        request_iterator: Iterator[SetBoundsParam], context: Any
    ) -> Empty:
        time.sleep(GRPC_SERVER_DELAY)
        return Empty()

    mocker.patch.object(servicer, "SetBounds", mock_set_bounds)
    server = MockGrpcServer(servicer, port=57809)
    await server.start()
```

and I expect the timeout error to be raised, but it doesn't happen (when I reduced the timeout to a very small number like `0.000001` it does work, but not always):
`await client.set_bounds(component_id=1, lower=0.0, upper=10.0, timeout=0.00001)`
Obviously, setting this timeout is not a good solution, but I am just trying to figure out why it behaves like this... I added a comment and asked if any of you have an idea what can be the case here.

Reply:
As discussed with @sahas-subramanian-frequenz, I've removed the test for testing timeout of the `SetBounds` method (because it also involves establishing a connection to the remote gRPC server and it cannot be easily tested, unless someone has an idea how to test it).